packages/react/src/components/Select/index.tsx

```diff
@@ -14,19 +14,74 @@ export interface SelectProps
   label: string;
   options?: SelectOption[];
   children?: React.ReactElement<HTMLOptionElement | HTMLOptGroupElement>[];
+  value?: any;
+  defaultValue?: any;
+  onChange: (e: React.ChangeEvent<HTMLSelectElement>) => void;
 }
 
 const Select = React.forwardRef(
   (
-    { options, children, disabled, label, id, required, ...rest }: SelectProps,
+    {
+      options,
+      children,
+      disabled,
+      label,
+      id,
+      required,
+      value,
+      defaultValue,
+      // eslint-disable-next-line @typescript-eslint/no-empty-function
+      onChange = () => {},
+      ...rest
+    }: SelectProps,
     ref: Ref<HTMLSelectElement>
   ): React.ReactElement<HTMLSelectElement> => {
     if (options && children) {
       console.error(
         'The Select component only takes the options props or child option elements, not both.'
       );
     }
+
+    const isControlled = typeof value !== 'undefined';
```

Comment:
This shouldn't hold up the PR in any way as I'm just thinking out loud: Do we want to account for `null` values?

Reply:
> Do we want to account for `null` values?

Good question. When in doubt I check and see what other react libraries do. I can follow up with a new ticket to address this once I do some research.